internal/pkg/runtime/engines/imgbuild/engine_linux.go

```diff
@@ -91,6 +92,13 @@ func (e *EngineOperations) PrepareConfig(starterConfig *starter.Config) error {
 		starterConfig.SetHybridWorkflow(true)
 		starterConfig.SetAllowSetgroups(true)
 
+		starterConfig.SetTargetUID(0)
+		starterConfig.SetTargetGID([]int{0})
+	} else if parent, setgroups := namespaces.ParentUserNamespace(os.Getpid()); parent && setgroups {
+		// we run in a parent user namespace and if setgroups is allowed
+		// we assume this process can fake root user, otherwise let it
```

Comment:
the "otherwise let ..." part...

does it mean that it _will_ fail unless we do this, or that it might still fail even if we do this?